fix(build): apply macOS entitlements via optionsForFile (#41)
holepunchto/hello-pear-electron: README.md

Note `APPLE_PASSWORD` is not the sign-in password, it's an [app-specific password][apple-app-specific-password].

`build/entitlements.mac.plist` declares the permissions required by the app under macOS Hardened Runtime, which is mandatory for notarized apps. The defaults cover Bare native addon compatibility:

- `cs.allow-jit` — required for V8/Bare JIT compilation
- `cs.allow-unsigned-executable-memory` — required for the Bare runtime

> Add or remove entitlements here if your app requires additional capabilities (e.g. camera, microphone, location). If your app loads native addons that dynamically link shared libraries built by a different developer (e.g. Homebrew-provided libraries), you will also need `cs.disable-library-validation`.

Reviewer: ```suggestion
Add or remove entitlements here  (e.g. camera, microphone, location). 

To load third-party native addons that dynamically link shared libraries built by a different developer  use `cs.disable-library-validation`.
```
